app/src/main/java/io/appium/uiautomator2/server/mjpeg/MjpegScreenshotStream.java

```diff
@@ -0,0 +1,142 @@
+import java.util.Iterator;
+import java.util.Locale;
+
+import io.appium.uiautomator2.model.internal.CustomUiDevice;
+import io.appium.uiautomator2.server.ServerConfig;
+import io.appium.uiautomator2.utils.ScreenshotHelper;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class MjpegScreenshotStream extends Thread {
+    private static final UiAutomation uia =
+        CustomUiDevice.getInstance().getInstrumentation().getUiAutomation();
+    private final List<MjpegScreenshotClient> clients;
+    private boolean stopped = false;
+
+    MjpegScreenshotStream(List<MjpegScreenshotClient> clients) {
+        this.clients = clients;
+    }
+
+    @Override
+    public void interrupt() {
+        this.stopped = true;
+        super.interrupt();
+    }
+
+    @Override
+    public void run() {
+        while (!stopped) {
+            if (clients.size() == 0) {
+                try {
+                    sleep(500);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+                continue;
+            }
+
+            // how long each loop should take in milliseconds
+            long targetInterval =
+                Math.round((1.0f / ServerConfig.getMjpegServerFramerate()) * 1000.0f);
+            long start = System.currentTimeMillis();
+
+            byte[] screenshotData = getScreenshot();
+            if (screenshotData.length > 0) {
+                MjpegScreenshotClient client;
+                for (Iterator<MjpegScreenshotClient> iterator = clients.iterator(); iterator.hasNext();) {
+                    client = iterator.next();
+                    if (client.getClosed()) {
+                        iterator.remove();
+                        continue;
+                    }
+
+                    if (!client.getInitialized()) {
+                        client.initialize();
+                    }
+
+                    client.write(screenshotData);
+                }
+            }
+            matchFramerate(targetInterval, start);
+        }
+    }
+
+    private void matchFramerate(long targetInterval, long start) {
+        long end = System.currentTimeMillis();
+        long duration = end - start;
+
+        if (duration < targetInterval) {
+            try {
+                sleep(targetInterval - duration);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private byte[] getScreenshot() {
+        Bitmap screenshot = uia.takeScreenshot();
+        byte[] jpeg = ScreenshotHelper.compressJpeg(
+            screenshot,
+            ServerConfig.getMjpegScalingFactor() / 100.0f,
+            ServerConfig.getMjpegServerScreenshotQuality(),
+            ServerConfig.getMjpegFiltering()
+        );
+        screenshot.recycle();
+
+        byte[] header = String.format(
+            Locale.ROOT,
+            "--BoundaryString\r\nContent-type: image/jpg\r\nContent-Length: %d\r\n\r\n",
```

Comment:
Please move the magic string to constants

Reply:
This will be resolved w/ the upcoming update(s)